### ISSUE TITLE
OpenMC MGXS Libraries are very slow to load

### ISSUE THREAD

**Issue by wbinventor** [mit-crpg/openmc#833]
I recently learned from @geogunow that it is now insanely slow to load MGXS libraries from a statepoint file (*i.e.*, the `openmc.mgxs.Library.load_from_statepoint(...)`). I've confirmed this with some recent tests with the BEAVRS model. My hunch is that I changed something in #807 which has led to this drastic slowdown, but I'm not sure. If anyone has noticed a major slowdown in simply loading tally data, please mention it here so we can work together to eliminate this issue.

**Comment by wbinventor** [comment]
The culprit much easier to track down than I expected, and originated with 7ca46e809ce01fe7857ae36072822a1718f01aaf. Evidently the `Geometry.get_all_*(...)` methods are now much, much slower than before. Rather than fix these methods, I will fix the way the `MGXS` class obtains its domain from the `Summary` object. I'll submit a PR for this shortly.

**Comment by paulromano** [comment]
Any idea why `get_all_*` is much slower?

**Comment by wbinventor** [comment]
I'm not entirely sure. Are dictionaries really that much slower to populate than lists?

**Comment by cjosey** [comment]
I'm not sure if this is helpful or not, but since I needed to know how much slower dictionaries were as well, I wrote this script: https://gist.github.com/cjosey/3b4acb408358746595df8d26983130fc

I modeled it after what I understood from the OpenMC code.  From this, lists are about half as fast.  If you remove sorting, the list code is now 3x faster.  So, either I've missed some form of complexity in the sample code, or it's not that.
